### DISCUSSION TITLE
qemu SITL

### DISCUSSION THREAD

**Discussion by LostBirdx** [nextpilot/nextpilot-flight-control#7] [Q&A]
I am very interested in this project, so I tried to run SITL on my machine. However, even after following all the steps mentioned in the documentation, I am unable to connect QGC to QEMU. I would like to know if the SITL functionality is currently working properly.

**Comment by AlexHAHA** [comment]
目前我们项目还在开发中，有很多模块还不能够完全运行。等初步能运行的时候给大家说哈。谢谢支持。
目前使用scons mavlink -j8可以看看mavlink的基本上下行联通测试



---- Replied Message ----
| From | Xiao ***@***.***> |
| Date | 06/23/2024 23:41 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [nextpilot/nextpilot-flight-control] qemu SITL (Discussion #7) |

I am very interested in this project, so I tried to run SITL on my machine. However, even after following all the steps mentioned in the documentation, I am unable to connect QGC to QEMU. I would like to know if the SITL functionality is currently working properly.

**Reply by ComerLater** [reply]
截止目前开发工具链、底层框架（uorb、param、workq）、飞行控制律、飞行模拟等大部分功能都已经移植，并编译通过。但全系统联调中，部分功能还在纯在bug，我们在边调试边修复bug。应该不会太久我们就能实现全系统在qemu上仿真，之后上px4 fmu-v5系列开源硬件。
